packages/elements/list/src/getListDeleteForward.ts

```diff
@@ -22,15 +30,52 @@ export const getListDeleteForward = (editor: SPEditor) => {
   const res = getListItemEntry(editor, {});
 
   let moved: boolean | undefined = false;
-  if (!isSelectionAtBlockEnd(editor) || !res) {
+  if (!isSelectionAtBlockEnd(editor)) {
     return moved;
   }
 
+  if (!res) {
+    const afterPoint = Editor.after(editor, editor.selection!.focus.path);
+
+    if (afterPoint) {
+      // there is a block after it
+      const nextSiblingListRes = getListItemEntry(editor, {
+        at: afterPoint.path,
+      });
+
+      if (nextSiblingListRes) {
+        // the next block is a list
+        const { listItem } = nextSiblingListRes;
+        const parentBlockEntity = getBlockAbove(editor, {
+          at: editor.selection!.anchor,
+        });
+
+        if (!getText(editor, parentBlockEntity![1])) {
+          // the selected block is empty
+          Transforms.removeNodes(editor, { at: editor.selection!.focus });
```

Comment:
Not sure if we should avoid asserting selection as non-null. Is there any case where it could be null or is it safe to leave as-is? Just making sure.

Reply:
Both valid, omitting at and checking for selection. Maybe the former, as it's a well-known thing in slate that the default location is selection, so it should be good. Up to you.

Reply:
In the open source context I'm generally ok with any time saver (e.g. type `any`) for new features, the priority being accepting any kind of contributions adding value. I took a lot of shortcuts I'll come back later on (thinking to typing).

Sure the next priority is to leverage strict typing, generic types and make Plate more robust.

Reply:
I'll omit the options part and also I'll put a check for `editor.selection!` at the beginning of `getListDeleteForward`